src/FastPriorityQueue/LegacyImplementation.php

```diff
@@ -27,69 +35,108 @@
  * the elements. This behaviour can be used in mixed scenarios with high
  * performance boost.
  */
-class FastPriorityQueue implements Iterator, Countable, Serializable
+class LegacyImplementation implements Iterator, Countable, Serializable
```

Comment:
Should be `@internal` IMO: the implementation should not be part of the public API